DeepSpeech.py

```diff
@@ -1169,6 +1213,10 @@ def _next_epoch(self):
                     # The current epoch should also have a validation part
                     self._epochs_running.append(Epoch(self._epoch, self._num_jobs_dev, set_name='dev', report=is_display_step))
 
+                    # State Machine flag for decision making regarding early stopping in next epoch function call
+                    self._dev = True
```

Comment:
Why in the next epoch? What if the next epoch is not a validating epoch? I'd place all early stopping code just here. Then there is also no need for the `self._dev`at all. The only state you'd need is `self._loss`.